### PR TITLE
JAMES-3428 Implement read-repairs for mailbox counters

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Comment by chibenwa** [comment]
`[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.2:test (default-test) on project apache-james-mailbox-elasticsearch: There was a timeout or other error in the fork -> [Help 1]`

unrelated?

**Comment by chibenwa** [comment]
test this please

File: mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageMapperTest.java [hunk 1]
```diff
@@ -353,4 +356,63 @@ void addShouldRetryMessageDenormalization(CassandraCluster cassandra) throws Exc
...
+            // 100 poll with a 0.1 probability to trigger read repair
+            Awaitility.await()
+                .pollInterval(new Duration(10, MILLISECONDS))
+                .atMost(Duration.ONE_SECOND)
+                .untilAsserted(() ->
+                    assertThat(messageMapper.getMailboxCounters(benwaInboxMailbox).getUnseen()).isEqualTo(4));
+        }
+
+        @Test
+        void readingShouldFixInvalidCounters(CassandraCluster cassandra) throws MailboxException {
```
Comment (Arsnael) [inline mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageMapperTest.java:403]: I'm not sure to get it between this test and the first one `readingShouldEventuallyFixCountersInconsistencies`

`readingShouldEventuallyFixCountersInconsistencies` => we create an inconsistency, do 100 reads with the fact that it does not happen everytime and check it got solved
`readingShouldFixInvalidCounters` => we create an inconsistency, one read and it's fixed? 

Or did I miss something?

Comment (chibenwa) [reply mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageMapperTest.java:403]: If the counter is invalid, we know for sure we have a consistency issue, we can fix it straight away, and return **directly** the right answer.

Probabilistic read repairs are probabilistic: when the data looks OK, we have say 1 chance out of 10, to perform a repair.

**Review by Arsnael — APPROVED** [review]

**Review by LanKhuat — APPROVED** [review]

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**